utility/src/main/java/org/oppia/android/util/networking/NetworkConnectionUtil.kt

```diff
@@ -1,18 +1,21 @@
 package org.oppia.android.util.networking
 
+import androidx.annotation.StringRes
+import org.oppia.android.util.R
+
 /** Utility to get the current connection status of the device. */
 interface NetworkConnectionUtil {
 
   /** Enum to distinguish different connection statuses for the device. */
-  enum class ProdConnectionStatus : ConnectionStatus {
+  enum class ProdConnectionStatus(@StringRes var connectionName: Int) : ConnectionStatus {
     /** Connected to WIFI or Ethernet. */
-    LOCAL,
+    LOCAL(connectionName = R.string.network_connection_local),
 
     /** Connected to Mobile or WiMax. */
-    CELLULAR,
+    CELLULAR(connectionName = R.string.network_connection_cellular),
 
     /** Not connected to a network. */
-    NONE
+    NONE(connectionName = R.string.network_connection_none)
```

Comment:
@BenHenning Added string support.

Reply:
These shouldn't be Android strings--they aren't shown in UIs (and generally utility shouldn't reference app strings as it super complicates the build graph).

Reply:
Updated to hardcode string